src/main/java/listeners/actions/SampleBlockActionListener.java

```diff
@@ -7,47 +7,58 @@
-import com.slack.api.methods.response.views.ViewsUpdateResponse;
 import com.slack.api.model.view.View;
 import com.slack.api.util.json.GsonFactory;
 import java.io.IOException;
-import java.util.Map;
+import models.Constants;
+import models.MeetingArrangement;
 
 public class SampleBlockActionListener implements BlockActionHandler {
+    private final App app;
+    private final Gson gson;
+
+    public SampleBlockActionListener(App app) {
+        this.app = app;
+        this.gson = GsonFactory.createSnakeCase(app.getSlack().getConfig());
+    }
 
     @Override
     public Response apply(BlockActionRequest req, ActionContext ctx) throws IOException, SlackApiException {
-        BlockActionPayload payload = req.getPayload();
-
-        String categoryId = payload.getActions().get(0).getSelectedOption().getValue();
-        View currentView = payload.getView();
-
-        String privateMetadata = currentView.getPrivateMetadata();
-        View viewForTheCategory = buildViewByCategory(categoryId, privateMetadata);
-
-        ViewsUpdateResponse viewsUpdateResp = ctx.client().viewsUpdate(r -> r.viewId(currentView.getId())
-                .hash(currentView.getHash())
-                .view(viewForTheCategory));
+        this.app.executorService().submit(() -> {
+            // run the main logic asynchronously
+            var payload = req.getPayload();
+            var categoryId = payload.getActions().get(0).getSelectedOption().getValue();
+            var currentView = payload.getView();
 
-        if (!viewsUpdateResp.isOk()) {
-            ctx.logger.error(viewsUpdateResp.toString());
-        }
+            var privateMetadata = currentView.getPrivateMetadata();
+            var viewForTheCategory = buildViewByCategory(categoryId, privateMetadata);
 
+            try {
+                var viewsUpdateResponse = ctx.client().viewsUpdate(r -> r.viewId(currentView.getId())
+                        .hash(currentView.getHash())
+                        .view(viewForTheCategory));
+                if (!viewsUpdateResponse.isOk()) {
+                    ctx.logger.error(viewsUpdateResponse.toString());
+                }
+            } catch (Exception e) {
+                ctx.logger.error("Failed to call views.update API (error: {})", e.getMessage(), e);
+            }
+        });
+        // immediately acknowledge the request
         return ctx.ack();
     }
 
-    View buildViewByCategory(String categoryId, String privateMetadata) {
-        Gson gson = GsonFactory.createSnakeCase();
-        Map<String, String> metadata = gson.fromJson(privateMetadata, Map.class);
-        metadata.put("categoryId", categoryId);
-        String updatedPrivateMetadata = gson.toJson(metadata);
-        return view(view -> view.callbackId("meeting-arrangement")
+    private View buildViewByCategory(String categoryId, String privateMetadata) {
+        var metadata = gson.fromJson(privateMetadata, MeetingArrangement.class);
```

Comment:
unchecked Map conversation results in warnings in the build process. instead, we can load the value as a class instance.